tools/bumper/parser_test.go

```diff
@@ -0,0 +1,208 @@
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Testing internal parser", func() {
+	var tmpfile *os.File
+
+	createTempFileWithContent := func(content []byte) {
+		var err error
+		tmpfile, err = ioutil.TempFile("", "test")
+		Expect(err).ToNot(HaveOccurred(), "Should successfully create temp file")
+
+		defer tmpfile.Close()
+
+		By("Writing and closing the temporary file")
+		_, err = tmpfile.Write(content)
+		Expect(err).ToNot(HaveOccurred(), "Should successfully write content to the temp file")
+	}
+
+	AfterEach(func() {
+		By("Removing the temporary file")
+		err := os.Remove(tmpfile.Name())
+		Expect(err).ToNot(HaveOccurred(), "Should successfully remove the temp file")
+	})
+
+	checkValidComponentsYaml := func(componentsConfig *componentsConfig) {
+		By("Checking Multus params are valid")
+		Expect(componentsConfig.Components["multus"].Url).To(Equal("https://github.com/intel/multus-cni"))
+		Expect(componentsConfig.Components["multus"].Commit).To(Equal("bfaf22964b51b93b08dcb1f10cbd8f8cb9a3603a"))
+		Expect(componentsConfig.Components["multus"].Metadata.Tag).To(Equal("v3.4.1"))
+
+		By("Checking LinuxBridge params are valid")
+		Expect(componentsConfig.Components["linux-bridge"].Url).To(Equal("https://github.com/containernetworking/plugins"))
+		Expect(componentsConfig.Components["linux-bridge"].Commit).To(Equal("ad10b6fa91aacd720f1f9ab94341a97a82a24965"))
+		Expect(componentsConfig.Components["linux-bridge"].Metadata.Tag).To(Equal("v0.8.6"))
+
+		By("Checking BridgeMarker params are valid")
+		Expect(componentsConfig.Components["bridge-marker"].Url).To(Equal("https://github.com/kubevirt/bridge-marker"))
+		Expect(componentsConfig.Components["bridge-marker"].Commit).To(Equal("049eb6a796b742767a0be838f6ddb7f79c97dd60"))
+		Expect(componentsConfig.Components["bridge-marker"].Metadata.Tag).To(Equal("0.3.0"))
+
+		By("Checking Kubemacpool params are valid")
+		Expect(componentsConfig.Components["kubemacpool"].Url).To(Equal("https://github.com/k8snetworkplumbingwg/kubemacpool"))
+		Expect(componentsConfig.Components["kubemacpool"].Commit).To(Equal("7728339b6c8dcfc9d53cc18ce8e06a910094ed47"))
+		Expect(componentsConfig.Components["kubemacpool"].Metadata.Tag).To(Equal("v0.16.0"))
+
+		By("Checking Nmstate params are valid")
+		Expect(componentsConfig.Components["nmstate"].Url).To(Equal("https://github.com/nmstate/kubernetes-nmstate"))
+		Expect(componentsConfig.Components["nmstate"].Commit).To(Equal("6f41747788235c3363a314dfae46a4ed90e1846a"))
+		Expect(componentsConfig.Components["nmstate"].Metadata.Tag).To(Equal("v0.25.0"))
+
+		By("Checking OvsCni params are valid")
+		Expect(componentsConfig.Components["ovs-cni"].Url).To(Equal("https://github.com/kubevirt/ovs-cni"))
+		Expect(componentsConfig.Components["ovs-cni"].Commit).To(Equal("edb4754d08f49be54c399c938895fe82aab6aa5a"))
+		Expect(componentsConfig.Components["ovs-cni"].Metadata.Tag).To(Equal("v0.12.0"))
```

Comment:
Gomega do a DeepEquals so is safe to compare the maps meaning that you can just compare input and output config instead of "unrolling" it, although maybe is not as clear at error we can fix that doing at gomega custom matcher for Component type.

Reply:
Changed to using an "expected" static struct. there is no need for a matcher imo the output is understandable
Done